### PR TITLE
feat(skymp5-client): taffy exping system, anim debug, feats

### PR DESCRIPTION
**Pull request by kkEngine**
- Changed animation debug system settings
- Added simple exping system
- Taffy perk system
- Implemented setTimeout, clearTimeout, setInterval, clearInterval

### CODE REVIEW

**Review by SupAidmi — APPROVED** [review]

File: skymp5-client/src/sync/inventory.ts [hunk 1]
```diff
@@ -324,6 +325,9 @@ export const applyInventory = (
 
   diff.sort((a, b) => (a.count < b.count ? -1 : 1));
   diff.forEach((e, i) => {
+
```
Comment (nic11) [inline skymp5-client/src/sync/inventory.ts:328]: We don't usually put newlines at the beginning of a block

File: skymp5-client/src/extensions/timers.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
+export { };
+import * as sp from "skyrimPlatform";
+
+//#region Timers declaration and implementation
```
Comment (nic11) [inline skymp5-client/src/extensions/timers.ts:4]: Wow, is it shown in vscode?

File: skymp5-client/src/features/authSystem.ts [hunk 3]
```diff
@@ -182,7 +183,7 @@ const checkLoginState = () => {
         default:
           ++browserState.failCount;
           browserState.comment = `Server returned ${response.status.toString() || "???"} "${response.body || response.error}"`;
-          sp.Utility.wait(1.5 + Math.random() * 2).then(checkLoginState);
+          setTimeout(() => checkLoginState(), 1.5 + Math.random() * 2);
```
[the remark this answers is not in the document]
Comment (kkEngine) [reply skymp5-client/src/features/authSystem.ts:186]: maybe but standard in ts/js is to use setTimeout

[the remark this answers is not in the document]
Comment (kkEngine) [reply skymp5-client/src/features/authSystem.ts:186]: i assume that the Utility class is using papyrus which is an extra load to CPU

File: skymp5-client/src/extensions/timers.ts (hunk 2 above)
Comment (kkEngine) [reply skymp5-client/src/extensions/timers.ts:4]: Yes, u can collapse region

File: skymp5-client/src/features/authSystem.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (nic11) [reply skymp5-client/src/features/authSystem.ts:186]: Hmm, yeah, that's probably right

File: skymp5-client/src/extensions/timers.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  }
+
+  updateEventHandle = sp.on("update", () => processTimers());
+});
+
+try {
+  if (sp.Game.getPlayer()!) { };
+  updateEventHandle = sp.on("update", () => processTimers());
+}
+catch {
```
Comment (nic11) [inline skymp5-client/src/extensions/timers.ts:135]:
```suggestion
} catch (e) {
  // also log something
```

File: skymp5-client/src/extensions/timers.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+sp.on("preLoadGame", () => {
+  if (updateEventHandle) {
+    sp.unsubscribe(updateEventHandle);
+  }
+
+  updateEventHandle = sp.on("update", () => processTimers());
+});
+
+try {
+  if (sp.Game.getPlayer()!) { };
```
Comment (nic11) [inline skymp5-client/src/extensions/timers.ts:132]: Isn't `!` only working as a static TS check?

File: skymp5-client/src/extensions/timers.ts (hunk 4 above)
Comment (kkEngine) [reply skymp5-client/src/extensions/timers.ts:135]: No log needed. I didn't find a way to check which context we are currently in. The reason for that is js hot reload.

Comment (kkEngine) [reply skymp5-client/src/extensions/timers.ts:135]: Hmm I just thought I could save in the storage the game state (main menu or game).

File: skymp5-client/src/extensions/timers.ts (hunk 5 above)
Comment (kkEngine) [reply skymp5-client/src/extensions/timers.ts:132]: Yes it is. It is better than
```
sp.Game.getPlayer() as Actor
```
As far as i know Game.getPlayer() can't return null according this page https://www.creationkit.com/index.php?title=GetPlayer_-_Game

Comment (kkEngine) [reply skymp5-client/src/extensions/timers.ts:132]: It seems that we have wrong definition in skyrimPlatform.ts

Comment (nic11) [reply skymp5-client/src/extensions/timers.ts]: But why is it needed here anyway?

File: skymp5-client/src/extensions/timers.ts [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+      return;
+    default:
+      break;
+  }
+
+  try {
+    if (sp.Game.getPlayer()!) { };
+    setProcessMethod(ProcessMethodType.update);
+  }
+  catch {
```
Comment (nic11) [inline skymp5-client/src/extensions/timers.ts:136]:
```suggestion
  } catch {
```
At least this way. (But I still don't understand why don't we want to log anything)

File: skymp5-client/src/sweetpie/playerCombatSystem.ts [hunk 7]
```diff
@@ -103,3 +105,81 @@ const getTimings = (weapon?: sp.WeaponType): [number, number] => {
...
+      enter: ((ctx) => {
+        if (playerLastStaminaValue < (staminaAttackMap.get("Power") ?? 0)) {
+          ctx.animEventName = "";
+        }
+      }),
+      leave: (() => { }),
+    }, 0x14, 0x14, pattern);
+  }
+
+  for (const pattern of ['JumpDirectionalStart*', 'JumpStandingstart*']) {
```
Comment (SupAidmi) [inline skymp5-client/src/sweetpie/playerCombatSystem.ts:149]: JumpStandingStart

**Review by nic11 — APPROVED** [review]

**Pull request merged**